lib/temporal/workflow/dispatcher.rb

```diff
@@ -26,7 +35,7 @@ def handlers_for(target, event_name)
         handlers[target]
           .concat(handlers[TARGET_WILDCARD])
           .select { |(name, _)| name == event_name || name == WILDCARD }
-          .map(&:last)
+          .sort_by { |(name, _)| name == WILDCARD ? 1 : 0 }
```

Comment:
After looking at the existing specs, sorting wildcard handlers to the end is probably not the behavior we want for non-query use cases?

Reply:
Can you help me understand why you need to sort handlers in the first place?

Reply:
For sure - the "catch-all" query handler registers itself with the wildcard `*` event name. This lets you use both targeted and catch-all query handlers within the same workflow (such as in the example workflow). The sorting here would reorder the wildcard catch-all handler to the end so that a specific handler takes precedence.